jwql/jwql_monitors/monitor_cron_jobs.py

```diff
@@ -0,0 +1,311 @@
+#! /usr/bin/env python
+
+"""This module monitors the status of JWQL monitors run via cron job via log files. Basic
+results (e.g. success, failure) are collected and placed in a Bokeh table for display
+on the web app
+
+Authors
+-------
+
+    - Bryan Hilbert
+
+Use
+---
+
+    This module can be executed as such:
+
+    ::
+
+        from jwql.jwql_monitors import monitor_cron_jobs
+        monitor_cron_jobs.status()
+
+Dependencies
+------------
+
+    The user must have a configuration file named ``config.json``
+    placed in the ``utils`` directory.
+"""
+
+from datetime import datetime
+import logging
+import numpy as np
+import os
+import time
+
+from bokeh.embed import components
+from bokeh.io import output_file, save, show
+from bokeh.layouts import widgetbox
+from bokeh.models import ColumnDataSource
+from bokeh.models.widgets import DataTable, DateFormatter, HTMLTemplateFormatter, TableColumn
+
+from jwql.utils.logging_functions import configure_logging, log_info, log_fail
+from jwql.utils.permissions import set_permissions
+from jwql.utils.utils import get_config
+
+
+def create_table(status_dict):
+    """Create interactive Bokeh table containing the logfile status results.
+
+    Parameters
+    ----------
+    status_dict : dict
+        Nested dictionary with status results from all logfiles
+    """
+    # Rearrange the nested dictionary into a non-nested dict for the table
```

Comment:
@bhilbert4 Another matter of preference thing, but another way to do this could be:

```python
filenames = [status_dict[monitor]['logname'] for monitor in status_dict]
dates = [datetime.fromtimestamp(status_dict[monitor]['latest_time']) for monitor in status_dict]
etc.
```

Reply:
Wouldn't that be slower, since it would involve 4 for loops instead of 1?

Reply:
Probably yes, but since we are talking about a small number of items in each list (probably tens of monitors when this project is fully built) I doubt that speed will be an issue at this point in the code.  Like I said, its a matter of preference in this case (slightly faster speed vs 4 lines of code instead of 9). 🤷‍♂️

Reply:
It would be interesting to time profile the use case, I should do that ...